primitives/arithmetic/src/traits.rs

```diff
@@ -367,14 +369,15 @@ mod ensure {
 		/// assert_eq!(overflow(), Err(ArithmeticError::Overflow));
 		/// assert_eq!(underflow(), Err(ArithmeticError::Underflow));
 		/// ```
-		fn ensure_add(self, v: Self) -> Result<Self, ArithmeticError> {
-			self.checked_add(&v).ok_or_else(|| error::equivalent(v))
+		fn ensure_add(mut self, v: Self) -> Result<Self, ArithmeticError> {
+			self.ensure_add_assign(v)?;
```

Comment:
From a reader's point of view, this seems weird, could it be possible to copy the `ensure_add_assign` implementation directly without the mutation?:
```rust
self.checked_add(&v).ok_or_else(|| error::equivalent(&v))?;
```

Reply:
This does not change the API. It is still callable on non-mutable objects.  

It could be done, but then I have code duplication and cannot use one Trait to test the other, since they could be independently wrong.

Reply:
Ok, you beat me this time 😝